web/src/main/java/org/cbioportal/web/ClinicalDataEnrichmentController.java

```diff
@@ -0,0 +1,102 @@
+import org.cbioportal.model.ClinicalDataEnrichment;
+import org.cbioportal.model.Sample;
+import org.cbioportal.service.ClinicalAttributeService;
+import org.cbioportal.service.SampleService;
+import org.cbioportal.web.config.annotation.InternalApi;
+import org.cbioportal.web.parameter.GroupFilter;
+import org.cbioportal.web.parameter.Projection;
+import org.cbioportal.web.parameter.SampleIdentifier;
+import org.cbioportal.web.util.ClinicalDataEnrichmentUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+@InternalApi
+@RestController
+@Validated
+@Api(tags = "Clinical Data Enrichments", description = " ")
+public class ClinicalDataEnrichmentController {
+
+    @Autowired
+    private ClinicalDataEnrichmentUtil clinicalDataEnrichmentUtil;
+
+    @Autowired
+    private ClinicalAttributeService clinicalAttributeService;
+
+    @Autowired
+    private SampleService sampleService;
+
+    @PreAuthorize("hasPermission(#groupFilter, 'GroupFilter', 'read')")
+    @RequestMapping(value = "/clinical-data-enrichments/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch clinical data enrichments for the sample groups")
+    public ResponseEntity<List<ClinicalDataEnrichment>> fetchClinicalEnrichments(
+            @ApiParam(required = true, value = "List of altered and unaltered Sample/Patient IDs") @Valid @RequestBody GroupFilter groupFilter) {
+
+        List<String> studyIds = groupFilter.getGroups().stream()
+                .flatMap(group -> group.getSampleIdentifiers().stream().map(SampleIdentifier::getStudyId))
+                .collect(Collectors.toList());
+
+        List<String> sampleIds = groupFilter.getGroups().stream()
+                .flatMap(group -> group.getSampleIdentifiers().stream().map(SampleIdentifier::getSampleId))
+                .collect(Collectors.toList());
+
+        List<Sample> samples = sampleService.fetchSamples(studyIds, sampleIds, Projection.SUMMARY.name());
+
+        MultiKeyMap sampleSet = new MultiKeyMap();
+
+        samples.stream()
+                .forEach(sample -> sampleSet.put(sample.getCancerStudyIdentifier(), sample.getStableId(), sample));
+
+        List<List<Sample>> groupedSamples = groupFilter.getGroups().stream()
```

Comment:
For non-trivial nested data structures (lists, maps, etc.) maybe we should add some comments to explain what they actually represent.